packages/platform-sdk-btc-indexer/src/database.ts

```diff
@@ -57,14 +52,18 @@ export class Database {
 	 * @returns {number}
 	 * @memberof Database
 	 */
-	public lastBlockNumber(): number {
-		const lastBlock = this.#database.prepare("SELECT height FROM blocks ORDER BY height DESC LIMIT 1").get();
-
-		if (lastBlock === undefined) {
+	public async lastBlockNumber(): Promise<number> {
+		const lastBlockHeight = await this.#prisma.block.aggregate({
+			_max: {
+				height: true,
+			},
+		});
+
+		if (lastBlockHeight === undefined) {
 			return 1;
 		}
 
-		return lastBlock.height;
+		return (lastBlockHeight["_max"]?.height as number) || 1;
```

Comment:
Type assertion unneeded here
```suggestion
		return lastBlockHeight["_max"]?.height || 1;
```